docs/_posts/2020-02-28-puppetwash.md

```diff
@@ -0,0 +1,283 @@
+```
+
+Let's take a look at what certificates were created
+```
+wash docker/containers > cd ../volumes/pupperware_puppetserver-config
+wash docker/volumes/pupperware_puppetserver-config > tree
+.
+├── auth.conf
+├── hiera.yaml
+├── puppet.conf
+├── puppetdb.conf
+└── ssl
+    ├── ca
+    │   ├── ca_crl.pem
+    │   ├── ca_crt.pem
+    │   ├── ca_key.pem
+    │   ├── ca_pub.pem
+    │   ├── infra_crl.pem
+    │   ├── infra_inventory.txt
+    │   ├── infra_serials
+    │   ├── inventory.txt
+    │   ├── requests
+    │   ├── serial
+    │   └── signed
+    │       ├── puppet.test.pem
+    │       └── puppetdb.test.pem
+    ├── certificate_requests
+    ├── certs
+    │   ├── ca.pem
+    │   └── puppet.test.pem
+    ├── crl.pem
+    ├── private
+    ├── private_keys
+    │   └── puppet.test.pem
+    └── public_keys
+        └── puppet.test.pem
+
+9 directories, 20 files
+```
+
+> Note that on Linux it'd be possible to find the volume with `docker inspect`, but on macOS Docker containers run in an isolated Linux VM.
+
+We can see that the CA has signed two certificates, one for `puppet.test` and another for `puppetdb.test`. We can see how PuppetDB expects to be addressed by looking at its Subject name
+```
+wash docker/volumes/pupperware_puppetserver-config ❯ openssl x509 -text -in ssl/ca/signed/puppetdb.test.pem | grep -m1 Subject
+        Subject: CN=puppetdb.test
+```
+So we should be able to connect to PuppetDB using it's default port 8081 at `puppetdb.test`. We can confirm that with `cat puppetdb.conf`. To get the local port we'll use to connect to it - within the Pupperware project - run
+```
+$ docker-compose port puppetdb 8081
+0.0.0.0:32770
+```
+
+Let's copy the certificates somewhere we can use them later. The Puppet Server will be in PuppetDB's whitelist, so let's use that one since it's easily accessible.
+```
+wash docker/volumes/pupperware_puppetserver-config ❯ mkdir -p ~/.puppetlabs/wash/pupperware
+wash docker/volumes/pupperware_puppetserver-config ❯ cp certs/ca.pem certs/puppet.test.pem ~/.puppetlabs/wash/pupperware/
+wash docker/volumes/pupperware_puppetserver-config ❯ cp private_keys/puppet.test.pem ~/.puppetlabs/wash/pupperware/puppet.test.key.pem
+```
+
+Last, let's make sure there's something in PuppetDB to look at. In the Pupperware project we'll do a noop agent run to add facts and a report
+```
+$ docker-compose exec puppet puppet agent -t --noop
+Info: Using configured environment 'production'
+Info: Retrieving pluginfacts
+Info: Retrieving plugin
+Info: Retrieving locales
+Info: Applying configuration version '1582848507'
+Info: Creating state file /opt/puppetlabs/puppet/cache/state/state.yaml
+Notice: Applied catalog in 0.02 seconds
+```
+
+## Configure Puppetwash
+
+To configure Puppetwash we need to install and configure it. It's distributed as a Ruby Gem, so we can install it with
+```
+gem install puppetwash
+```
+
+Get the path to the Puppetwash script with
+```
+$ gem contents puppetwash
```

Comment:
`gem contents puppetwash --version <version>` ? Should probably update puppetwash README to be similar to boltwash.